leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/CaliforniumLwM2mRequestSender.java

```diff
@@ -47,6 +51,7 @@
     private final Set<Endpoint> endpoints;
     private final ObservationRegistry observationRegistry;
     private final LwM2mModelProvider modelProvider;
+    private final Map<String/* registration id */, ArrayList<Request> /* pending coap Request */> pendingRequests = new HashMap<>();
```

Comment:
If the "pendingRequest" should not be "serialized", I think it's possible to exclude them.

May idea about this was very simple:
- the server already has a map with the <reg-id> as key. And the access to such a map is not well scaling.
- so just reuse that "already made access" by storing the stuff in the objects maintained by the first map.